### PR TITLE
PixelPaint: Implement Fit Canvas to Layers/Selection

### PR DESCRIPTION
**Pull request by Sestowner**

### CODE REVIEW

**Review by LucasChollet — COMMENTED** [review]
I don't know PixelPaint to say something on the logic, but here are some nitpicks on the code.

File: Userland/Applications/PixelPaint/Image.cpp [hunk 1]
```diff
@@ -528,6 +528,47 @@ void Image::crop(Gfx::IntRect const& cropped_rect)
...
 
+Optional<Gfx::IntRect> Image::bounding_rect() const
+{
+    if (m_layers.is_empty())
+        return {};
+
+    Optional<int> min_x;
+    Optional<int> min_y;
+    Optional<int> max_x;
+    Optional<int> max_y;
```
Comment (linusg) [inline Userland/Applications/PixelPaint/Image.cpp:539]: Let's use two `IntPoint`s for these instead

File: Userland/Applications/PixelPaint/MainWidget.cpp [hunk 2]
```diff
@@ -577,6 +577,29 @@ void MainWidget::initialize_menubar(GUI::Window& window)
...
+            auto* editor = current_image_editor();
+            VERIFY(editor);
+            auto bounding_rect = editor->image().bounding_rect();
+            if (!bounding_rect.has_value())
+                return;
+            editor->image().set_canvas_size(bounding_rect.value());
+            editor->did_complete_action("Fit Canvas to Layers"sv);
+        }));
+    m_image_menu->add_action(GUI::Action::create(
+        "&Fit Canvas to Selection", [&](auto&) {
```
Comment (linusg) [inline Userland/Applications/PixelPaint/MainWidget.cpp:592]: You can't have the same `&` letter for two actions

File: Userland/Applications/PixelPaint/MainWidget.cpp [hunk 3]
```diff
@@ -577,6 +577,29 @@ void MainWidget::initialize_menubar(GUI::Window& window)
...
+            auto* editor = current_image_editor();
+            VERIFY(editor);
+            // FIXME: disable this action if there is no selection
+            if (editor->image().selection().is_empty())
+                return;
+            auto rect = editor->image().selection().bounding_rect();
+            editor->image().set_canvas_size(rect);
+            editor->image().selection().clear();
```
Comment (linusg) [inline Userland/Applications/PixelPaint/MainWidget.cpp:600]: Why?

**Review by linusg — CHANGES_REQUESTED** [review]

File: Userland/Applications/PixelPaint/MainWidget.cpp (hunk 3 above)
Comment (Sestowner) [reply Userland/Applications/PixelPaint/MainWidget.cpp:600]: To be consistent with "Crop [Image](https://github.com/SerenityOS/serenity/blob/d5cf403f709a39e2a5b436d149d86b6eef1571b0/Userland/Applications/PixelPaint/MainWidget.cpp#L591)/[Layer](https://github.com/SerenityOS/serenity/blob/d5cf403f709a39e2a5b436d149d86b6eef1571b0/Userland/Applications/PixelPaint/MainWidget.cpp#L857) to Selection".

File: Userland/Applications/PixelPaint/Image.cpp (hunk 1 above)
Comment (Sestowner) [reply Userland/Applications/PixelPaint/Image.cpp:539]: What do you think about this?
```
Optional<Gfx::IntRect> Image::bounding_rect() const
{
    if (m_layers.is_empty())
        return {};

    Optional<Gfx::IntRect> rect;

    for (auto const& layer : m_layers)
        rect = rect.has_value() ? rect->united(layer.relative_rect()) : layer.relative_rect();

    return rect;
}
```

File: Userland/Applications/PixelPaint/MainWidget.cpp (hunk 2 above)
Comment (Sestowner) [reply Userland/Applications/PixelPaint/MainWidget.cpp:592]: It is now possible to [cycle through menu items with the same Alt shortcut](https://github.com/SerenityOS/serenity/commit/288d75fdabf161cc47d7326d63334702969673ae).

**Pull request closed without merging**